File: src/main/java/com/vinyl/model/Release.java
package com.vinyl.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Release {
	private String releaseBarcode;
	private String albumCatalogNum;
	private String releaseCountryCode;
	private Date releaseDate;
	private Integer recordSize;
	private Integer recordSpeed;
	private Integer copiesCount;
	private Boolean isRepress;
	private String label;
	private List<String> recordBarcodes;
}

File: src/main/resources/frontend/src/schemas/release.js
export default {
  props: {
    releaseBarcode: 'string',
    albumCatalogNum: {
      type: 'album',
      isSchema: true
    },
    releaseCountryCode: 'country',
    releaseDate: 'date',
    recordSize: '7|10|12',
    recordSpeed: '33|45|78',
    copiesCount: 'int',
    isRepress: 'boolean',
    label: {
      type: 'string',
      isNullable: true
    }
  },
  key: 'releaseBarcode',
  display: (item) => `${item.albumCatalogNum} ${item.releaseDate} ${item.label}`
}